sources/packages/backend/libs/integrations/src/services/sfas/sims-to-sfas.service.ts

```diff
@@ -119,6 +127,128 @@ export class SIMSToSFASService {
+      ELSE NULL
+     END`,
+          "applicationCancelDate",
+        )
+        .innerJoin("application.currentAssessment", "studentAssessment")
+        .innerJoin("application.programYear", "programYear")
+        .innerJoin("application.student", "student")
+        .innerJoin("studentAssessment.offering", "offering")
+        .leftJoin(
+          "studentAssessment.disbursementSchedules",
+          "disbursementSchedule",
+        )
+        .leftJoin(
+          "disbursementSchedule.disbursementValues",
+          "disbursementValues",
+        )
+        .where("application.applicationStatus != :overwritten")
+        // Check if the application data was updated in the given period.
+        .andWhere(
+          "application.updatedAt > :modifiedSince AND application.updatedAt <= :modifiedUntil",
+        )
+        .setParameters({
+          overwritten: ApplicationStatus.Overwritten,
+          cancelled: ApplicationStatus.Cancelled,
+          modifiedSince,
+          modifiedUntil,
+        })
+        .groupBy("application.id")
+        .addGroupBy("student.id")
+        .addGroupBy("programYear.id")
+        .addGroupBy("studentAssessment.id")
+        .addGroupBy("offering.id")
+        .orderBy("offering.offeringIntensity")
+        .getRawMany<ApplicationRecord>()
+    );
+  }
+
+  /**
+   * Get all student ids and restriction data of students who have one or more updates in restriction related data.
+   * @param modifiedSince the date after which the restriction data was updated.
+   * @param modifiedUntil the date until which the restriction data was updated.
+   */
+  async getAllRestrictionWithUpdates(
+    modifiedSince: Date,
+    modifiedUntil: Date,
+  ): Promise<RestrictionRecord[]> {
+    return (
+      this.restrictionRepo
+        .createQueryBuilder("studentRestriction")
+        .select("studentRestriction.id", "restrictionId")
+        .addSelect("student.id", "studentId")
+        .addSelect("restriction.restrictionCode", "restrictionCode")
+        .addSelect("studentRestriction.createdAt", "restrictionEffectiveDate")
+        // Conditionally set restrictionRemovalDate based on isActive status.
+        .addSelect(
+          `CASE 
+      WHEN studentRestriction.is_active = false THEN studentRestriction.updatedAt 
+      ELSE NULL 
+     END`,
+          "restrictionRemovalDate",
+        )
+        .innerJoin("studentRestriction.student", "student")
+        .innerJoin("studentRestriction.restriction", "restriction")
+        // Check if the restriction data was updated in the given period.
+        .where("restriction.restrictionType = :restrictionType")
+        .andWhere(
+          "studentRestriction.updatedAt > :modifiedSince AND studentRestriction.updatedAt <= :modifiedUntil",
+        )
+        .setParameters({
+          restrictionType: RestrictionType.Provincial,
+          modifiedSince,
+          modifiedUntil,
+        })
+        .getRawMany<RestrictionRecord>()
```

Comment:
I would highly recommend this to be a type orm regular query and if possible the typeorm.find().